### PR TITLE
Vet clinic database: database performance audit

### PR DESCRIPTION
**Pull request by Feliverse**
Preparation (
Run the following query to add an extra column to the owners table:
```
-- Add an email column to your owners table
ALTER TABLE owners ADD COLUMN email VARCHAR(120);
```
Run the following statements to add data to your database (executing them might take a few minutes):
```
-- This will add 3.594.280 visits considering you have 10 animals, 4 vets, and it will use around ~87.000 timestamps (~4min approx.)
INSERT INTO visits (animal_id, vet_id, date_of_visit) SELECT * FROM (SELECT id FROM animals) animal_ids, (SELECT id FROM vets) vets_ids, generate_series('1980-01-01'::timestamp, '2021-01-01', '4 hours') visit_timestamp;

-- This will add 2.500.000 owners with full_name = 'Owner <X>' and email = 'owner_<X>@email.com' (~2min approx.)
insert into owners (full_name, email) select 'Owner ' || generate_series(1,2500000), 'owner_' || generate_series(1,2500000) || '@mail.

The following queries are taking too much time (1 sec = 1000ms can be considered as too much time for database query). Try them on your machine to confirm it:

```
- `SELECT COUNT(*) FROM visits where animal_id = 4;`
- `SELECT * FROM visits where vet_id =SELECT * FROM owners where email = 'owner_18327@mail.com'; 2;`
- ` SELECT * FROM owners where email = 'owner_18327@mail.com';`

BEFORE:
image

AFTER APPLYING INDEX TO INCREASE THE PERFORMANCE :
image

BEFORE:
image

AFTER APPLYING INDEX TO INCREASE THE PERFORMANCE :
image

BEFORE:
image

AFTER APPLYING INDEX TO INCREASE THE PERFORMANCE :
image

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -153,3 +153,22 @@ SELECT species.name, count(*)
...
+ 
+ EXPLAIN ANALYZE SELECT COUNT(*) FROM visits where animal_id = 4;
+ EXPLAIN ANALYZE SELECT * FROM visits where vet_id = 2;
+ EXPLAIN ANALYZE SELECT * FROM owners where email = 'owner_18327@mail.com';
+ 
+ /*Apply Indexes to improbe the preformance*/
+ 
+ CREATE INDEX animals_id_index ON visits(animal_id ASC);
+ CREATE INDEX vets_id_index ON visits(vet_id, animal_id, date_of_visit);
+ CREATE INDEX email_index ON owners(email ASC);
```
Comment (franclobo) [inline queries.sql]: - [x] Well done implementing the solution to low the execution time, however, one of the basic solutions for the performance problems is adding indexes for FOREIGN KEYS or on columns frequently used in queries. Please add the `CREATE INDEX` lines on the schema.sql file. 👇🏽 

> Everything that creates or change the tables structure should be saved in the schema.sql file.
> Everything that populates tables with data should be saved in the data.sql file.
> Everything that is a query that helps to analyze data should be saved in the queries.sql file.

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @Feliverse and @francis3400, 

# ♻️ Required Changes ♻️

so-close-jim-halpert

Good job so far!
You are doing it well 👏🏽 there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- Correct git flow ✔️ 
- Descriptive PR ✔️ 
- Correct use of `EXPLAIN ANALYZE` 💯 
- Good use of CREATE INDEX 💯 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 1 above)
Comment (Feliverse) [reply queries.sql]: Ok, I will change, thank you

**Review by Munsa1 — APPROVED** [review]
Hi @Feliverse, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Implemented Required Change :heavy_check_mark: 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**